src/QueryBuilder.js

```diff
@@ -1566,6 +1570,229 @@ QueryBuilder.prototype.returning = queryMethod('returning');
 QueryBuilder.prototype.truncate = queryMethod('truncate');
 
 /**
+ * Json query APIs
+ */
+
+/**
+ * Json field filter.
+ *
+ * Supports having field expression in both sides of operand, but right hand value cannot
+ * be any other value, but only string reference to other column/json filed, json array
+ * or json object.
+ *
+ * Converts left and right hand values to PostgreSQL acceptable format and add user chosen
+ * operator between left and right hand expressions.
+ *
+ * @param fieldExpression {String} Reference to column / jsonField.
+ * @param jsonObjectOrFieldExpression {Object|Array|String} Reference to column / jsonField or json object.
+ * @returns {QueryBuilder}
+ */
+QueryBuilder.prototype.whereJsonObject = function (fieldExpression, operator, jsonObjectOrFieldExpression) {
+  var fieldReference = parseFieldExpression(fieldExpression);
+  if (_.isString(jsonObjectOrFieldExpression)) {
+    var rightHandReference = parseFieldExpression(jsonObjectOrFieldExpression);
+    var refRefQuery = ["(", fieldReference, ")::jsonb ", operator, " (", rightHandReference, ")::jsonb"].join("");
+    return this.whereRaw(refRefQuery);
+  } else if (_.isObject(jsonObjectOrFieldExpression)) {
+    var refValQuery = ["(", fieldReference, ")::jsonb ", operator, " ?::jsonb"].join("");
+    return this.whereRaw(refValQuery, JSON.stringify(jsonObjectOrFieldExpression));
+  }
+  throw new Error("Invalid right hand expression.");
+};
+
+
+/**
+ * Json equality comparison.
+ *
+ * Also supports having field expression in both sides of equality.
+ *
+ * ```js
+ * Person
+ *   .query()
+ *   .whereJsonEquals('additionalData.myDogs', 'additionalData.dogsAtHome')
+ *   .then(function (person) {
+ *     // oh joy! these persons has all their dogs at home!
+ *   });
+ *
+ * Person
+ *   .query()
+ *   .whereJsonEquals('additionalData.myDogs[0]', { name: "peter"})
+ *   .then(function (person) {
+ *     // these persons' first dog name is "peter" and the dog has no other
+ *     // attributes, but its name
+ *   });
+ * ```
+ *
+ * @param fieldExpression {String} Reference to column / jsonField.
+ * @param jsonObjectOrFieldExpression {Object|Array|String} Reference to column / jsonField or json object.
+ * @returns {QueryBuilder}
+ */
+QueryBuilder.prototype.whereJsonEquals = function (fieldExpression, jsonObjectOrFieldExpression) {
+  return this.whereJsonObject(fieldExpression, "=", jsonObjectOrFieldExpression);
+};
+
+/**
+ * Json filters all results where left hand operator is superset of the right hand operand.
+ *
+ * ```js
+ * Person
+ *   .query()
+ *   .whereJsonSupersetOf('additionalData.myDogs', 'additionalData.dogsAtHome')
```

Comment:
Isn't `additionalData.myDogs` interpreted so that `additionalData` is the table name and `myDogs` the column? Shouldn't this be `additionalData:myDogs`?